torch/_prims_common/__init__.py

```diff
@@ -442,19 +442,22 @@ def validate_exclusive_idx(rank: int, ex_idx: int):
 
 # "Wraps" a dim (up to one time) for the given rank, allowing
 # dims to be specified using negative indices
-def canonicalize_dim(rank: int, idx: int) -> int:
-    # TODO: add a comment for why this is
-    _rank = rank if rank != 0 else 1
-
-    if idx >= 0 and idx < _rank:
+def canonicalize_dim(rank: int, idx: int, wrap_scalar: bool = True) -> int:
+    if rank <= 0:
```

Comment:
rank can never ever be less than 0

Reply:
while passing data from the proper apis, it cannot. but perhaps this might be used in some testing code where ranks are generated (incorrectly). so i'd rather keep this to catch issues early (it's an int here after all). i don't think there's any penalty for having this vs `== 0`.

Reply:
I agree there's no penalty, but I think it would make sense to assert that rank is non-negative, to catch errors early instead of masking them by resetting rank to 1.